File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int b=0;
        int cnt=0;
        for(char c:s)
        {
            if(c=='(')
            b++;
            else if(c==')')
            {
                b--;
                
            }

            if(b<0)
            {cnt++;
            b=0;
            }


            
        }

        return abs(cnt)+abs(b);
    }
};

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        b=0
        cnt=0
        for c in s:
            if c=='(':
                b+=1
            else:
                b-=1
            if b<0:
                cnt+=1
                b=0
        return abs(b)+abs(cnt)
        